module/VuFind/src/VuFind/ILS/Driver/AbstractBase.php

```diff
@@ -66,6 +68,56 @@ public function setConfig($config)
         $this->config = $config;
     }
 
+    /**
+     * Create a patron array. Ensures that all the keys are present in the resulting array.
+     *
+     * @param string  $id           Patron id in catalog
+     * @param string  $cat_username Catalog username, usually the same as barcode
+     * @param string  $cat_password Catalog password
+     * @param ?string $email        Patron email in catalog
+     * @param string  $firstname    Patron first name
+     * @param string  $lastname     Patron last name
+     * @param string  $barcode      Obtained barcode, if empty default to cat_username
```

Comment:
Barcode is not a documented field in the [spec](https://vufind.org/wiki/development:plugins:ils_drivers). Is there a reason to add it here and introduce this defaulting behavior, or would it be better to treat the drivers that reference barcode as having an extra field? If we don't change the default barcode behavior, we can avoid a number of unit test changes in this PR. I don't see any existing code that makes use of the patron barcode value, and if in future we do want to use barcodes, it's probably better to ensure that the field really contains a barcode, rather than a potentially arbitrary username value.

Reply:
I've now added a parameter: nonDefaultFields to the array creation functions, in which those are assigned. And removed all the values which were not documented in their corresponding specs.